### ISSUE TITLE
Submit to Kodi's repo

### ISSUE THREAD

**Issue by PanderMusubi** [mpolednik/script.kodi.hue.ambilight#30]
Is this now the leading repo for https://github.com/michaelrcarroll/script.kodi.hue.ambilight ?

If so, which code and issues need to move over to here?

At least this issue can be worked at https://github.com/michaelrcarroll/script.kodi.hue.ambilight/issues/24

Let me know if you need help submitting it to Kodi's repo. Having it added there will greatly improve the awareness of users and developers that this add-on exists.

**Comment by Moyster** [comment]
Hello,

I'm not the repo owner but I remember @mpolednik accepted to be upstream.

I don't think any code needs to be fetched from @michaelrcarroll 's repo, I have no idea for the issue tickets.
https://github.com/michaelrcarroll/script.kodi.hue.ambilight/issues/24 is right about users increase, and @mpolednik recently said he was working on refactoring some parts of the code to clean the plugin and allow more granularity when it comes to configuring the lights (behavior).
So, it's all up to him to decide if this plugin as of now deserves to be added or if he prefers to wait a bit.

@mpolednik : how about rounding to v 0.8 considering the recent updates ? (lightstrips, v17, translations, etc..)

**Comment by JesCR** [comment]
i think this is the leading repo for kodi.hue...

And the move to the official repo is in issue #19 

Great Work!!!

**Comment by mpolednik** [comment]
@Moyster yeah version bump could be possible (although I'd prefer people to be just running git master for now).

As for the inclusion: having this in official repo is a great milestone. My problem is that current code base has few issues (e.g. bad indentation) and no tests to do simple automatic conversion without possibility of breaking stuff. If we manage to fix that, I'm totally up for bumping to 1.0 and getting in touch with Kodi submission process.

**Comment by JesCR** [comment]
If you need my set up for testing something...
I've two hue bloom for ambilight, and two bulbs that i want to use as theatre... but i don't know how to group...
Settings ony allow 3 lights...

**Comment by PanderMusubi** [comment]
Testing abd grouping can be done with https://github.com/craigcabrey/luminance

Where do you need help for indentation?

**Comment by mpolednik** [comment]
For the testing, I was referring to unit tests -- I don't think we need functional or more advanced tests. My biggest concern for indentation changes is that it could very subtly modify behavior of critical code path:

if condition:
    f()
    g()

VS

if condition:
    f()
g()

that we wouldn't easily detect without unit tests. After we have some test coverage, it's mostly trivial just to use tool such as https://pypi.python.org/pypi/autopep8 to fix the indentation.

**Comment by mpolednik** [comment]
Sorry, I can't github. :) 

    if condition:
        f()
        g()

VS

    if condition:
        f()
    g()
